### ISSUE TITLE
callables as default values

### ISSUE THREAD

**Issue by kevinsl** [prawn-cake/simple-models#3]
Hello, thanks for this module! It's very interesting. I'm using it in a project now and trying to extend it. I'm not sure I understand how a field's default values are supposed to work when they are specified as callable types (I'm using `datetime.now`). Here's an example of my problem:

``` python
from datetime import datetime
from simplemodels.fields import SimpleField
from simplemodels.models import Document
import time


class Message(Document):
    msg = SimpleField(required=True)
    ts = SimpleField(required=True, default=datetime.now)


message1 = Message(msg='first')

time.sleep(2)

message2 = Message(msg='second')

print message1
print message2
```

I'd expect message1 and message2 to have ts fields which are 2 seconds off. But printing their contents shows the timestamps are the same.

```
{'msg': 'first', 'ts': datetime.datetime(2015, 10, 9, 7, 58, 45, 358320)}
{'msg': 'second', 'ts': datetime.datetime(2015, 10, 9, 7, 58, 45, 358320)}
```

Do you have any suggestions here? thanks

**Comment by prawn-cake** [comment]
Hi, thanks for the feedback.
This is good point, it doesn't work as expected.
Fortunately it's easy to fix. I've just released new version 0.3.1 with this fix and some other updates.
Check it out.

**Comment by kevinsl** [comment]
Thanks! That works but now I am running into another issue. Taking the above code, if I add a validator to `ts` field like `, validators=[datetime])` I get a validation error. It seems it's trying to pass the datetime.now method into the validator.

**Comment by prawn-cake** [comment]
Unfortunately you can't use datetime as a validator. Validator is assumed to be a callable object, function with only one parameter, but `datetime` expect multiple parameters: year, month, day, etc.

Solution is wrap it to a custom validation function, for instance:

```
def is_dt(value):
    if not isinstance(value, datetime):
        raise ValidationError('Value is not a datetime instance')

class Message(Document):
    ts = SimpleField(default=datetime.now, validators=[is_dt])
```

You also can use class as a validator, but it looks redundant in this case

```
class DatetimeValidator(object):
    def __init__(self, value):
        if not isinstance(value, datetime):
            raise ValidationError()
```

But, perhaps it might be reasonable to implement some builtin Validator class with additional options, for example for such `is_instance` validations

**Comment by kevinsl** [comment]
Okay, thanks but I'm still having trouble. I've made changes like you suggested but with the custom validator it's receiving `value` as a method, not datetime. Here my code: https://gist.github.com/kevinsl/361171c96713fa9b2b48

**Comment by kevinsl** [comment]
Actually I just upgraded to 0.3.2 and it's working. Previously I had been using 0.3.1 but installed manually because it wasn't in pypi yet.

I'll keep working with this and let you know how it goes. thanks!

**Comment by prawn-cake** [comment]
Yeah, sorry, I forgot to upload 0.3.1 and released 0.3.2 version soon.
Look at last develop, there I added experimental `is_instance` validator function which is useful for your case. 
Check it [here](https://github.com/prawn-cake/simple-models/blob/develop/simplemodels/utils.py#L6)

Example:

```
class Message(Document):
    ts = SimpleField(default=datetime.now, validators=[is_instance(datetime)])
```

Yeah, thanks for the feedback! Enjoy =)
